src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -649,5 +649,31 @@ public async Task<IEnumerable<ReviewModel>> GetReviewsAsync(string ServiceName,
         {
             return await _reviewsRepository.GetReviewsAsync(ServiceName, PackageName, filterType);
         }
+
+        public async Task AutoArchiveReviews(int archiveAfterMonths)
+        {
+            var reviews = await _reviewsRepository.GetReviewsAsync(false, "All", filterType: ReviewType.Manual);
+            // Find all inactive reviews
+            reviews = reviews.Where(r => r.LastUpdated.AddMonths(archiveAfterMonths) < DateTime.Now);
+            foreach (var review in reviews)
+            {
+                var requestTelemetry = new RequestTelemetry { Name = "Archiving Review " + review.ReviewId };
+                var operation = _telemetryClient.StartOperation(requestTelemetry);
+                try
+                {
+                    review.IsClosed = true;
+                    await _reviewsRepository.UpsertReviewAsync(review);
+                    await Task.Delay(500);
```

Comment:
What is the delay here for?

Reply:
Delaying background task to iterate after delay to avoid taking most of the processing just for the background task so user requests won't get impacted.